registration-processor/pre-processor/registration-processor-introducer-validator-stage/src/main/java/io/mosip/registration/processor/stages/introducer/IntroducerValidator.java

```diff
@@ -0,0 +1,308 @@
+import io.mosip.registration.processor.core.exception.RegistrationProcessorCheckedException;
+import io.mosip.registration.processor.core.exception.util.PlatformErrorMessages;
+import io.mosip.registration.processor.core.logger.RegProcessorLogger;
+import io.mosip.registration.processor.core.status.util.StatusUtil;
+import io.mosip.registration.processor.core.util.RegistrationExceptionMapperUtil;
+import io.mosip.registration.processor.packet.manager.idreposervice.IdRepoService;
+import io.mosip.registration.processor.packet.storage.utils.AuthUtil;
+import io.mosip.registration.processor.packet.storage.utils.PriorityBasedPacketManagerService;
+import io.mosip.registration.processor.packet.storage.utils.Utilities;
+import io.mosip.registration.processor.stages.utils.StatusMessage;
+import io.mosip.registration.processor.status.code.RegistrationStatusCode;
+import io.mosip.registration.processor.status.dto.InternalRegistrationStatusDto;
+import io.mosip.registration.processor.status.dto.RegistrationStatusDto;
+import io.mosip.registration.processor.status.dto.SyncTypeDto;
+import io.mosip.registration.processor.status.service.RegistrationStatusService;
+
+@Service
+public class IntroducerValidator {
+
+	private static Logger regProcLogger = RegProcessorLogger.getLogger(IntroducerValidator.class);
+
+	public static final String INDIVIDUAL_TYPE_UIN = "UIN";
+
+	@Autowired
+	RegistrationExceptionMapperUtil registrationExceptionMapperUtil;
+
+	@Autowired
+	private PriorityBasedPacketManagerService packetManagerService;
+
+	@Autowired
+	private IdRepoService idRepoService;
+
+	@Autowired
+	RegistrationStatusService<String, InternalRegistrationStatusDto, RegistrationStatusDto> registrationStatusService;
+
+	@Autowired
+	ObjectMapper mapper;
+
+	@Autowired
+	private AuthUtil authUtil;
+
+	@Autowired
+	private Utilities utility;
+
+	@Value("${mosip.kernel.applicant.type.age.limit}")
+	private String ageLimit;
+
+	@Value("${registration.processor.applicant.dob.format}")
+	private String dobFormat;
+
+	@Value("${registration.processor.validate.introducer}")
+	private boolean introducerValidation;
+
+	/**
+	 * Checks if is valid introducer.
+	 *
+	 * @param registrationId        the registration id
+	 * @param registrationStatusDto
+	 * @return true, if is valid introducer
+	 * @throws IOException                                Signals that an I/O
+	 *                                                    exception has occurred.
+	 * @throws SAXException
+	 * @throws ParserConfigurationException
+	 * @throws NoSuchAlgorithmException
+	 * @throws InvalidKeySpecException
+	 * @throws BiometricException
+	 * @throws io.mosip.kernel.core.exception.IOException
+	 * @throws BaseCheckedException
+	 * @throws PacketDecryptionFailureException
+	 * @throws RegistrationProcessorCheckedException
+	 */
+	public boolean isValidIntroducer(String registrationId, InternalRegistrationStatusDto registrationStatusDto)
```

Comment:
Since this stage does not do multiple validations like CMD stage, all these logic to be in IntroducerValidationProcessor class as specificed in the Jira task. Do create it as a separate method. Please recheck the details in task description with the code once for all the 4 stages

Reply:
I see this is not taken care, any reasons ?

Reply:
there are 5 methods present in introducerValidator class and more than 200 line of code... that's why....